script.js

```diff
@@ -136,18 +206,40 @@ my_widget_script =
         //LA calls this method with b_suppress_message and relies on your code to communicate issues to the user
         //Returning an empty array [] or NULL equals no error
         //TO DO write code specific to your form
-     
         return this.parent_class.is_valid(b_suppress_message);
       },
-     
+
       is_edited:function () {
         //should return true if the form has been edited since it was loaded or since reset_edited was called
         return this.parent_class.is_edited();
       },
-     
+
       reset_edited:function () {
         //typically called have a save
         //TO DO write code specific to your form
         return this.parent_class.reset_edited();
-      }
-    }
+      },
+
+      // Enable all records if the first record fields are correctly filled
+      enable_all_records: function() {
+        console.log('**** In enable_all_records ****')
+        $('#the_form tbody tr:not(.initialRow) input').prop('disabled', false)
```

Comment:
if you want to leave moles disabled, I think this can become something like:
```suggestion
        $('#the_form tbody tr:not(.initialRow) input:not([name^=moles])').prop('disabled', false)
```